pennylane/_grad.py

```diff
@@ -332,3 +334,130 @@ def _jacobian_function(*args, **kwargs):
         return jac[0] if unpack else jac
 
     return _jacobian_function
+
+
+# pylint: disable=too-many-arguments
+def vjp(f, params, cotangents, method=None, h=None, argnum=None):
```

Comment:
```suggestion
def vjp(f, params, cotangents, method=None, step_size=None, argnum=None):
```

Reply:
I don't agree that we should change the name of an argument, we widely use h for the step size for finite difference. https://docs.pennylane.ai/en/stable/code/api/pennylane.gradients.finite_diff.html?highlight=finite%20dif#pennylane.gradients.finite_diff

Reply:
We initially used `h` in Catalyst as well but found users didn't necessarily know what that meant. In general this sort of variable name is also discouraged since it is only meaningful in a certain context/background.
Here it might be more important to stick with the existing PL convention though. @josh146 @trbromley

Reply:
I'm happy with whatever decision we make here. But, we should be consistent with the name of step-size for all gradients methods: `grad`, `jacobian`, `vjp`, and `jvp` that we have implemented Catalyst's patchers.

Reply:
Ah, I just had a look and we tend to use `h` everywhere:

- [`catalyst.grad`](https://docs.pennylane.ai/projects/catalyst/en/latest/code/api/catalyst.grad.html)
- [`catalyst.jacobian`](https://docs.pennylane.ai/projects/catalyst/en/latest/code/api/catalyst.jacobian.html)
- [`catalyst.vjp`](https://docs.pennylane.ai/projects/catalyst/en/latest/code/api/catalyst.vjp.html)
- [`catalyst.jvp`](https://docs.pennylane.ai/projects/catalyst/en/latest/code/api/catalyst.jvp.html)
- [`qml.gradients.finite_diff`](https://docs.pennylane.ai/en/stable/code/api/pennylane.gradients.finite_diff.html)

As a result, I tend to favour `h` for consistency for now. (but agree that `step_size` is better, maybe we consider changing in the future).

@maliasadi could you update the PR https://github.com/PennyLaneAI/pennylane/pull/4709 to use `h`?